### DISCUSSION TITLE
Which parameter values in the dictionary are actually used by a given model?

### DISCUSSION THREAD

**Discussion by MaricelaM** [pybamm-team/PyBaMM#1539] [Q&A] [answered]
Hello there! 

I was wondering if there is a way to print out or access information on which parameter values are actually used by a given model. For example, 

`model = pybamm.lithium_ion.DFN()` and `model =pybamm.lithium_ion.SPM()`

Produce the same result when calling `print(model.parameters)` produces the same number of parameters in both cases, as does  `Params = pybamm.ParameterValues(chemistry=pybamm.parameter_sets.Chen2020)`

` for key, value in params.items():`

> `       try:`

> > `            print(key, value)`

 

> `      except TyperError:`

  >>`            print(key, value.__srt__())`. 

However, less parameters are required to fully describe the SPM than the DFN. I’m wondering how to see only the parameters that are needed to describe a given model. 

Any guidance would be very appreciated! 

Best, 

Mari

**Comment by rtimms** [comment, marked as answer]
You can do 

```python3
import pybamm

model = pybamm.lithium_ion.DFN()
model.print_parameter_info()
```

to get a list of all the parameters that show up in the model.
